src/txn/txn_recover.c

```diff
@@ -237,7 +237,7 @@ __txn_op_apply(WT_RECOVERY *r, WT_LSN *lsnp, const uint8_t **pp, const uint8_t *
         WT_ERR(__wt_logop_row_remove_unpack(session, pp, end, &fileid, &key));
         GET_RECOVERY_CURSOR(session, r, lsnp, fileid, &cursor);
         __wt_cursor_set_raw_key(cursor, &key);
-        WT_ERR(cursor->remove(cursor));
+        WT_ERR_NOTFOUND_OK(cursor->remove(cursor), false);
```

Comment:
Did you see failures if this wasn't changed, that is, did tests fail, and if so, which ones?

Reply:
Yes, I was seeing some failures in test format (very easily reproducible if I remember correctly) where it is complaining that it fails to apply an operation during recovery due to WT_NOTFOUND.  I can dig up the patch if you'd like to see the specific failure message.

Reply:
I'm curious how we'd get there. For format to write a log record removing a non-existent record, it seems to me it would have to successfully remove a non-existent record, and that's not supposed to be possible any more, is it?

So, I don't understand why this change is necessary, and it concerns me.

Reply:
As Keith said, the above two modifications are for applying log operations that are succeeded. We don't expect that record miss during the log apply.  Better to remove these two changes.

Reply:
@mm-ng, I was unable to reproduce this problem, and I really think it shouldn't happen. So, I've backed out this change and pushed the branch, hoping testing will fail. Or, if you have a reproducer, please let me know.

I'll own chasing the problem as soon as we have a reproducer.